modules/security-lti/src/main/java/org/opencastproject/security/lti/LtiLaunchAuthenticationHandler.java

```diff
@@ -405,31 +425,27 @@ public Authentication createAuthentication(HttpServletRequest request, ConsumerA
    * @param userAuthorities
    *          Collection to append to.
    */
-  private void enrichRoleGrants(String roles, String context, Collection<GrantedAuthority> userAuthorities) {
-    // Roles could be a list
+  private void enrichRoleGrants(String roles, String context, final String rolePrefix,
+      Collection<GrantedAuthority> userAuthorities) {
     if (roles != null) {
+      // Roles could be a list
       String[] roleList = roles.split(",");
 
       // Use a generic context and learner if none is given:
       context = StringUtils.isBlank(context) ? DEFAULT_CONTEXT : context;
 
-      for (String learner : roleList) {
+      for (final String ltiRole : roleList) {
         // Build the role
-        String role;
-        String group;
-        if (learner.equals(customRoleName)) {
-          for (String rolename : customRoles) {
-            userAuthorities.add(new SimpleGrantedAuthority(rolename));
+        if (ltiRole.equals(customRoleName)) {
+          for (String roleName : customRoles) {
+            userAuthorities.add(new SimpleGrantedAuthority(roleName));
           }
         }
 
-        if (StringUtils.isBlank(learner)) {
-          role = context + "_" + DEFAULT_LEARNER;
-        } else {
-          role = context + "_" + learner;
-          group = "ROLE_GROUP_" + learner.toUpperCase();
-          logger.debug("Adding group: {}", group);
-        }
+        final String normalizedLtiRole = StringUtils.defaultIfBlank(ltiRole, DEFAULT_LEARNER);
+        final String role = rolePrefix + context + "_" + normalizedLtiRole;
+        final String group = "ROLE_GROUP_" + rolePrefix + normalizedLtiRole.toUpperCase();
+        logger.debug("Adding group: {}", group);
```

Comment:
The group is never added! Is anybody relying on having groups like `ROLE_GROUP_Learner`,
`ROLE_GROUP_Instructor`? I don't think this was advertised. And as far as I know, no role provider adds these roles.

I suggest to remove the two above lines and create a new PR if this feature is needed.

Reply:
> @lkiesow you had a simple LTI testing site right? Any hints on how to easily test this? Otherwise, I will set up a Moodle installation which I need anyway for the other Moodle PRs.

You are probably looking for
https://github.com/opencast/helper-scripts/tree/be75207b9ad20c2edae8efbd642adb291f965c32/lti-test-consumer

Reply:
Interesting. You are right. I just modified the old code where the group role is created:

https://github.com/opencast/opencast/blob/18e4e5f194e5c750fb2613af42d1e09491e9b4ed/modules/security-lti/src/main/java/org/opencastproject/security/lti/LtiLaunchAuthenticationHandler.java#L428-L432

But even here it's never actually used. Will remove it.